script/seed-github-enterprise-server-instance.ts

```diff
@@ -174,71 +179,256 @@ const createIssue = async ({
   return id;
 };
 
-const octokit = new Octokit({
-  auth: process.env.GITHUB_TOKEN,
-  baseUrl: process.env.GITHUB_BASE_URL,
-});
+const encryptSecretValue = ({
+  sodium,
+  key,
+  secretValue,
+}: {
+  sodium: typeof _sodium;
+  key: string;
+  secretValue: string;
+}): string => {
+  const binKey = sodium.from_base64(key, sodium.base64_variants.ORIGINAL);
+  const binSecretValue = sodium.from_string(secretValue);
+  const encryptedSecretValue = sodium.to_base64(
+    sodium.crypto_box_seal(binSecretValue, binKey),
+    sodium.base64_variants.ORIGINAL,
+  );
+
+  return encryptedSecretValue;
+};
+
+const encryptAndSetActionsSecret = async ({
+  octokit,
+  owner,
+  repo,
+  sodium,
+  secretName,
+  secretValue,
+}: {
+  octokit: OctokitType;
+  owner: string;
+  repo: string;
+  sodium: typeof _sodium;
+  secretName: string;
+  secretValue: string;
+}) => {
+  const {
+    data: { key_id: keyId, key },
+  } = await octokit.request('GET /repos/{owner}/{repo}/actions/secrets/public-key', {
+    owner: owner,
+    repo: repo,
+  });
+
+  const encryptedSecretValue = encryptSecretValue({ sodium, key, secretValue });
+
+  await octokit.request('PUT /repos/{owner}/{repo}/actions/secrets/{secret_name}', {
+    owner,
+    repo,
+    secret_name: secretName,
+    encrypted_value: encryptedSecretValue,
+    key_id: keyId,
+  });
+};
```

Comment:
It will fill in the `{secret_name}` placeholder in the path:

```ts
await octokit.request('PUT /repos/{owner}/{repo}/actions/secrets/{secret_name}', {
  // ...
});
```